src/api/spec/models/attrib_spec.rb

```diff
@@ -1,19 +1,106 @@
 require 'rails_helper'
 
-RSpec.describe Attrib do
+RSpec.describe Attrib, :type => :model do
   let(:attribute) { create(:attrib, project: create(:project)) }
   let(:project) { create(:project) }
   let(:package) { create(:package) }
 
+  context "#fullname" do
+    it { expect(attribute.fullname).to eq("#{attribute.namespace}:#{attribute.name}") }
+  end
+
+  context "#container" do
+    context "attribute with project" do
+      it { expect(attribute.container).to eq(attribute.project) }
+    end
+
+    context "attribute with package" do
+      let(:attribute_with_package) { create(:attrib, package: package) }
+
+      it { expect(attribute_with_package.container).to eq(package) }
+    end
+  end
+
   context "#container=" do
-    it "sets the project" do
-      attribute.container = project
-      expect(attribute.container).to be(project)
+    context "assigning a project" do
+      before do
+        attribute.container = project
+      end
+
+      it { expect(attribute.container).to be(project) }
+      it { expect(attribute.project).to be(project) }
+      it { expect(attribute.package).to be_nil }
+
+      context "and then assigning a package" do
```

Comment:
> I don't think that this is necessary! If assigning a package works then it also works when you assigned a project before, right?

It works, but I think it is necessary because the result of assigning a project and then a package is the same as assigning a package and then a project, and maybe it shouldn't.

I would expect that assigning a package and then a project results in a project. But no. For this last case it results in a package. Assigning a project after assigning a package makes nothing.